### PR TITLE
Dynamic loading error messages

### PR DESCRIPTION
**Pull request by permcody**
Fixing the bad error messages found in the dynamic loading routines and other messaging improvements.

### CODE REVIEW

File: framework/src/base/MooseApp.C
Comment (lindsayad) [inline framework/src/base/MooseApp.C:1812]:
```suggestion
    oss << '"' << app_name << "\" is not a registered application name.\n"
        << "No search paths were set. We made no attempts to locate the corresponding library "
           "file.\n";
```

File: modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i
Comment (lindsayad) [inline modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i:10]:
```suggestion
  [u]
  []
```

File: modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i
Comment (lindsayad) [inline modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i:21]:
```suggestion
  [diff]
    type = Diffusion
    variable = u
  []
  [td]
    type = TimeDerivative
    variable = u
  []
```

File: modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i
Comment (lindsayad) [inline modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i:36]:
```suggestion
  [left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
  [right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  []
```

File: modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i
Comment (lindsayad) [inline modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i:55]:
```suggestion
  [sub_app]
```

File: modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i
Comment (lindsayad) [inline modules/misc/test/tests/dynamic_loading/dynamic_load_multiapp/misc_parent_no_path.i:62]:
```suggestion
  []
```

File: python/TestHarness/testers/Tester.py
Comment (lindsayad) [inline python/TestHarness/testers/Tester.py:641]: Do we output the "problematic" variable that was set?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: python/TestHarness/testers/Tester.py
Comment (permcody) [reply python/TestHarness/testers/Tester.py:641]: This isn't trivial to change. All this caveat logic is checked before the test is run to know whether we should attempt to run the test. There is no output created or saved so we'd have to refactor the test harness and do something quite different here. I would suggest we don't worry about it because we are officially using this in one place now and only for a single variable that can be inspected from the 'spec file' if it is skipped.

**Review by loganharbour — CHANGES_REQUESTED** [review]
Need to hold off on merge until new coverage target is good (enable new changes on next right before merging)

**Review by lindsayad — APPROVED** [review]

**Comment by loganharbour** [comment]
FYI @lindsayad we decided to let the misc dynamic loading tests contribute to coverage, too. It's currently only on this PR. I'll change that, and then merge this.

https://mooseframework.inl.gov/docs/PRs/24774/coverage_diff/framework/ looks good. There's a few small things that we get false, non-framework additions from... but that's okay

**Review by loganharbour — APPROVED** [review]

**Pull request merged**
